src/components/TravelPlanCard.tsx

```diff
@@ -0,0 +1,166 @@
+  if (daysLeft > 0) return "bg-blue-500 text-white";
+  return "bg-gray-500 text-white";
+};
+
+export default function TravelPlanCard({
+  plan,
+  onEdit,
+  onDelete,
+}: TripCardProps) {
+  const { id, name, description, startDate, endDate } = plan;
+  const daysLeft = calculateDaysLeft(startDate);
+  const daysLeftText = getDaysLeftText(daysLeft);
+  const [isEditModalOpen, setIsEditModalOpen] = useState(false);
+  const dialog = document.getElementById("dialog");
+  dialog?.addEventListener("click", (e) => {
+    const target = e.target;
+    console.log(target);
+    const rect = target.getBoundingClientRect();
+    if (
+      rect.left > e.clientX ||
+      rect.right < e.clientX ||
+      rect.top > e.clientY ||
+      rect.bottom < e.clientY
+    ) {
+      setIsEditModalOpen(false);
+    }
+  });
+  return (
+    <Link
+      to={`/travel/${id}`}
+      state={{ trip: plan }}
+      className="block"
+      onClick={(e) => {
+        if (isEditModalOpen) e.preventDefault();
+      }}
+    >
+      <Card className="w-full relative hover:shadow-lg transition-shadow">
+        <div
+          className="absolute top-2 right-2 z-10 no-link"
+          onClick={(e) => e.stopPropagation()}
+        >
+          <Button
+            variant="ghost"
+            size="icon"
+            onClick={(e) => {
+              e.preventDefault();
+              setIsEditModalOpen(true);
+            }}
+          >
+            <Pencil className="h-4 w-4" />
+            <span className="sr-only">Edit trip</span>
+          </Button>
+          <Button
+            variant="ghost"
+            size="icon"
+            onClick={(e) => {
+              e.preventDefault();
+              e.stopPropagation();
+              onDelete(id);
+            }}
+          >
+            <X size={16} />
+            <span className="sr-only">delete trip</span>
+          </Button>
+        </div>
+        <div className="absolute top-2 left-2">
+          <Badge variant="secondary" className={getDaysLeftColor(daysLeft)}>
+            {daysLeftText}
+          </Badge>
+        </div>
+        <CardHeader className="pt-10">
+          <CardTitle>{name}</CardTitle>
+          <CardDescription>
+            {startDate} - {endDate}
+          </CardDescription>
+        </CardHeader>
+        <CardContent>
+          <p>{description}</p>
+        </CardContent>
+      </Card>
+      <div
+        id="dialog"
+        onClick={(e) => {
+          e.preventDefault();
+          e.stopPropagation();
+        }}
+      >
+        <Dialog
+          open={isEditModalOpen}
+          onOpenChange={(isOpen) => setIsEditModalOpen(isOpen)}
+        >
+          <DialogContent onClick={(e) => e.stopPropagation()}>
+            <DialogHeader>
+              <DialogTitle>여행 수정</DialogTitle>
+            </DialogHeader>
+            <EditTripForm
+              trip={plan}
+              onSave={(updatedTrip: TripPlan) => {
+                onEdit(updatedTrip);
+              }}
```

Comment:
```suggestion
              onSave={onEdit}
```

이렇게 써도 괜찮지 않을까요?
비슷한 다른 부분들도 한 번 확인해보시면 좋을 거 같아요~